src/main/java/com/browserstack/local/LocalBinary.java

```diff
@@ -40,15 +40,31 @@ private void initialize() throws LocalException {
         } else if (osname.contains("mac") || osname.contains("darwin")) {
             binFileName = "BrowserStackLocal-darwin-x64";
         } else if (osname.contains("linux")) {
-            String arch = System.getProperty("os.arch");
-            binFileName = "BrowserStackLocal-linux-" + (arch.contains("64") ? "x64" : "ia32");
+            if (isAlpine()) {
+                binFileName = "BrowserStackLocal-alpine";
+            } else {
+                String arch = System.getProperty("os.arch");
+                binFileName = "BrowserStackLocal-linux-" + (arch.contains("64") ? "x64" : "ia32");
+            }
         } else {
             throw new LocalException("Failed to detect OS type");
         }
 
         httpPath = BIN_URL + binFileName;
     }
 
+    private boolean isAlpine() {
+        String[] cmd = { "/bin/sh", "-c", "grep -w \"NAME\" /etc/os-release" };
+        try {
+            Process os = Runtime.getRuntime().exec(cmd);
+            BufferedReader stdout = new BufferedReader(new InputStreamReader(os.getInputStream()));
+
+            return stdout.readLine().contains("Alpine");
+        } catch (Exception e) {
+            return false;
+        }
```

Comment:
Possibility of encountering an Error?

`public InputStreamReader(InputStream in) {
        super(in);
        try {
            sd = StreamDecoder.forInputStreamReader(in, this, (String)null); // ## check lock object
        } catch (UnsupportedEncodingException e) {
            // The default encoding should always be available
            throw new Error(e);
        }
    }`